gamedata.fbs

```diff
@@ -0,0 +1,367 @@
+  last_input:ControllerState (required);
+  /// If the player was the last one to be watched by a spectator.
+  last_spectated:bool;
+  /// True if the player has jumped. See dodge_timeout to know if a dodge/secondary jump is available.
+  has_jumped:bool;
+  /// True if the player has doubled jumped.
+  has_double_jumped:bool;
+  /// True if the player has dodged.
+  has_dodged:bool;
+  /// The time in seconds since the last dodge was initiated.
+  /// Resets to 0 when the player lands on the ground.
+  dodge_elapsed:float;
+  /// The unit direction of the latest dodge.
+  // TODO: What if it was a stall?
+  dodge_dir:Vector2 (required);
+}
+
+/// Information about a ball.
+table BallInfo {
+  /// The physical state of the ball.
+  physics:Physics (required);
+  /// The collision shape of the ball.
+  shape:CollisionShape (required);
+}
+
+/// The state of a boost pad.
+/// Note, static properties of boost pads, such as their location and size, are found in the field info.
+struct BoostPadState {
+  /// True if the boost can be picked up right now.
+  is_active:bool;
+  /// The number of seconds since the boost has been picked up, or 0 if the boost is active.
+  /// A big boost pad becomes active again after 10 seconds.
+  /// A small boost pad becomes active again after 4 seconds.
+  timer:float;
+}
+
+/// Possible phases of the match.
+enum MatchPhase: ubyte {
+  /// Match has not been created yet.
+  Inactive,
+  /// 3-2-1 countdown of a kickoff.
+  Countdown,
+  /// After kickoff countdown, but before ball has been hit.
+  /// The match automatically proceeds to Active after 2 seconds.
+  Kickoff,
+  /// The ball is in play and time is ticking.
+  Active,
+  /// A goal was just scored. Waiting for replay to start.
+  GoalScored,
+  /// Goal replay is being shown.
+  Replay,
+  /// The match is paused.
+  Paused,
+  /// The match has ended.
+  Ended,
+}
+
+/// Information about the current match such as time and gravity.
+table MatchInfo {
+  /// How many seconds have elapsed since the first game packet of the match.
+  /// This value ticks up even during kickoffs, replays, pause, etc.
+  seconds_elapsed:float;
+  /// Seconds remaining of the match.
+  /// This value ticks up instead of down during overtime or when the game duration mutator is set to Unlimited.
+  /// I.e. it matches the in-game timer at the top.
+  game_time_remaining:float;
+  /// True if the game is in overtime.
+  is_overtime:bool;
+  /// True if the game duration is set to Unlimited.
+  is_unlimited_time:bool;
+  /// The current phase of the match, i.e. kickoff, replay, active, etc.
+  game_status:GameStatus;
+  /// The current strength of gravity. Default is -650.
+  world_gravity_z:float;
+  /// Game speed multiplier. Regular game speed is 1.0.
+  game_speed:float;
+  /// Tracks the number of physics frames the game has computed.
+  /// May increase by more than one across consecutive packets.
+  /// Data type will roll over after 207 days at 120Hz.
```

Comment:
207 days is actually for an int instead of a uint - a uint will roll over after 414 days! (2^32-1) / 120 / 60 / 60 / 24 = 414.252246817 and (2^31-1) / 120 / 60 / 60 / 24 = 207.12612336